sdk/data/azcosmos/cosmos_client.go

```diff
@@ -49,6 +50,42 @@ func NewClient(endpoint string, cred azcore.TokenCredential, o *ClientOptions) (
 	return &Client{endpoint: endpoint, pipeline: newPipeline(newCosmosBearerTokenPolicy(cred, scope, nil), o)}, nil
 }
 
+// NewClientWithConnectionString creates a new instance of Cosmos client with connection string. It uses the default pipeline configuration.
+// connectionString - The cosmos service connection string.
+// options - Optional Cosmos client options.  Pass nil to accept default values.
+func NewClientWithConnectionString(connectionString string, o *ClientOptions) (*Client, error) {
```

Comment:
Please update the doc.go file: https://github.com/Azure/azure-sdk-for-go/blob/main/sdk/data/azcosmos/doc.go#L27-L33